django/urls/resolvers.py

```diff
@@ -329,8 +334,22 @@ def check(self):
                     id="2_0.W001",
                 )
             )
+        warnings.extend(self._check_angle_bracket_pattern())
```

Comment:
You can move it to the first line, e.g.
```python
def check(self):
    warnings = [
        *self._check_pattern_startswith_slash(),
        *self._check_angle_bracket_pattern(),
    ]
    ...
```

Reply:
Nice! I think we can even refactor this check:
`        if "(?P<" in route or route.startswith("^") or route.endswith("$"):
            warnings.append(
                Warning(
                    "Your URL pattern {} has a route that contains '(?P<', begins "
                    "with a '^', or ends with a '$'. This was likely an oversight "
                    "when migrating to django.urls.path().".format(self.describe()),
                    id="2_0.W001",
                )
            )`

as a separate method like two others, and then just something like:
`
        warnings = [
            *self._check_pattern_startswith_slash(),
            *self._check_pattern_unmatched_angle_bracket(),
             self._check_pattern_caret_and_dollar()
        ]`

Then, it might look more consistent and more readable.

Reply:
> I think we can even refactor this check

I'm not sure it's worth doing. It's not related with this patch so don't do this here.

Reply:
Yes, I won't include it in this patch. Maybe can be discussed as a ticket. Probably, such things have appeared more than once in the project

Reply:
> Maybe can be discussed as a ticket.

Ticket is not required for small cleanups. In general, we try to avoid trivial code reformatting unless it's a big problem as it creates extra noise in the history.

Reply:
Aha! Got it. I try to get aligned!